types/usb/index.d.ts

```diff
@@ -42,7 +42,11 @@ export class Device {
   interfaces: Interface[];
 
   __open(): void;
+  __getConfigDescriptor(): ConfigDescriptor;
```

Comment:
TBF, I'm not sure about exposing these, but as they exist and `__open()` has already been exposed, I'm OK with it.

Reply:
Like I said, it's the only way to check the active configuration before selecting one.
This procedure is [in the official libusb documentation](http://libusb.sourceforge.net/api-1.0/libusb_caveats.html#configsel).

Reply:
I think the active configuration is available via:

```
<device>.configDescriptor.bConfigurationValue
```

Reply:
I thought so too; unfortunately `configDescriptor` is wrongly [cached on first access](https://github.com/tessel/node-usb/blob/cb760d799a0558c5160ba07874aceebcde495eaf/usb.js#L55). The issue is worse than it seems, because after a successful `setConfiguration`, the same (possibly cached) descriptor is [used to build the interfaces](https://github.com/tessel/node-usb/blob/cb760d799a0558c5160ba07874aceebcde495eaf/usb.js#L255).

So, right now, in order to perform the procedure you need to use `__getConfigDescriptor().bConfigurationValue` directly.

Reply:
I mean, the whole libusb API is very hazardous to use in cases where you need to manually configure / claim the device, but we'd need breaking API changes for a proper fix. That's why I suggest we expose these internals (so users can at least work around the issues in the meanwhile).

Reply:
Would be awesome if this could be fixed in `node-usb` too!

Reply:
That would be awesome, yes... I suggested a patch to them a while ago, let's see if they reply.